### PR TITLE
PRO-1040 WalletConnect Chain Check

### PR DESCRIPTION
**Pull request by hemadri-tank-bacancy**
Conditions for **eth_signedTypedData** and **eth_signedTypedData** call request.

1. If chainId not found then refused request and show error message.
2. If dApp chainId not match with call request then reject request and show error message.
3. If dApp chainId match with call request then show request modal.

### CODE REVIEW

File: src/locales/en/common.json [hunk 1]
```diff
@@ -584,7 +584,8 @@
       "alreadyPending": "A connection is already waiting",
       "callRequestRejectFailed": "Call Request reject failed",
       "callRequestApproveFailed": "Call Request approve failed",
-      "invalidRequest": "This WalletConnect request didn't look quite right - please try again",
+      "invalidRequest": "This WalletConnect request didn't look quite right - please try again ",
+      "cannotDetermineChain": "Stating that no Chain was found for the incoming request, and to contact {{ dAppName }} ",
```
Comment (IAmKio) [inline src/locales/en/common.json:588]:
```suggestion
      "cannotDetermineChain": "Sorry, but no chain information was found in this request. Please contact {{ dAppName }}",
```

**Review by IAmKio — CHANGES_REQUESTED** [review]
Just one small change, but other than that it looks good so far

File: src/locales/en/common.json (hunk 1 above)
Comment (hemadri-tank-bacancy) [reply src/locales/en/common.json]: Changed it

File: src/actions/walletConnectActions.js [hunk 2]
```diff
@@ -375,6 +377,23 @@ export const subscribeToWalletConnectConnectorEventsAction = (connector: WalletC
         return;
       }
 
+      if (method === ETH_SIGN_TYPED_DATA || method === ETH_SIGN_TYPED_DATA_V4) {
+        const { domain } = JSON.parse(params[1]);
```
Comment (IAmKio) [inline src/actions/walletConnectActions.js:381]: `JSON.parse` throws an error if there is an error in the JSON or something else like `params[index]` does not exist. Please ensure thta params[index] exists and / or wrap this in a try / catch and handle the error.

**Review by IAmKio — CHANGES_REQUESTED** [review]
Apologies just another thing i noticed here that could cause a crash

File: src/actions/walletConnectActions.js (hunk 2 above)
Comment (hemadri-tank-bacancy) [reply src/actions/walletConnectActions.js:386]: Well, I made one more condition

File: src/actions/walletConnectActions.js [hunk 3]
```diff
@@ -378,6 +379,10 @@ export const subscribeToWalletConnectConnectorEventsAction = (connector: WalletC
       }
 
       if (method === ETH_SIGN_TYPED_DATA || method === ETH_SIGN_TYPED_DATA_V4) {
+        if (isEmpty(params)) {
+          dispatch(setWalletConnectErrorAction(t('error.walletConnect.cannotDetermineChain', { dAppName: name })));
```
Comment (IAmKio) [inline src/actions/walletConnectActions.js:383]: Could we please add a sentry / error log here so we can track when this has happened. Thanks

**Review by IAmKio — APPROVED** [review]
LGTM

**Pull request merged**